src/http_rest_api_handler.cpp

```diff
@@ -668,14 +670,31 @@ Status HttpRestApiHandler::processListModelsRequest(std::string& response) {
     return StatusCode::OK;
 }
 
-Status HttpRestApiHandler::processV3(const std::string_view uri, const HttpRequestComponents& request_components, std::string& response, const std::string& request_body, std::shared_ptr<HttpAsyncWriter> serverReaderWriter, std::shared_ptr<MultiPartParser> multiPartParser) {
+Status HttpRestApiHandler::processV3(const std::string_view uri, const HttpRequestComponents& request_components, std::string& response, const std::string& request_body, std::shared_ptr<HttpAsyncWriter> serverReaderWriter, std::shared_ptr<MultiPartParser> multiPartParser, const std::string& api_key) {
 #if (MEDIAPIPE_DISABLE == 0)
     OVMS_PROFILE_FUNCTION();
 
     HttpPayload request;
     std::string modelName;
     bool streamFieldVal = false;
-
+    // convert headers to lowercase because http headers are case insensitive
+    for (const auto& [key, value] : request_components.headers) {
+        std::string lowercaseKey = key;
+        std::transform(lowercaseKey.begin(), lowercaseKey.end(), lowercaseKey.begin(), 
+                        [](unsigned char c){ return std::tolower(c); });
+    }
+    if (!api_key.empty()) {
+        if (request_components.headers.count("authorization")) {
+            if (request_components.headers.at("authorization") != "Bearer " + api_key) {
```

Comment:
```suggestion
            if (request_components.headers.at("authorization") != "Bearer " + apiKey) {
```
Is exact match safe here? No risk of some additional whitespaces?